olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -281,18 +281,27 @@ client::CancellationToken VersionedLayerClientImpl::PrefetchPartitions(
           }
         };
 
-        context.ExecuteOrCancelled([&]() {
-          auto download_job =
-              std::make_shared<PrefetchPartitionsHelper::DownloadJob>(
-                  std::move(download), std::move(append_result),
-                  std::move(call_user_callback), std::move(status_callback));
-          return PrefetchPartitionsHelper::Prefetch(
-              std::move(download_job), request.GetPartitionIds(),
-              std::move(query), task_sink_, kQueryPartitionsMaxSize,
-              request.GetPriority());
-        });
-
-        return EmptyResponse(client::ApiNoResult());
+        bool prefetch_triggered = true;
+
+        context.ExecuteOrCancelled(
```

Comment:
Use the return value instead of adding an additional lambda for achieving the same thing.

Same for below changes.